### PR TITLE
Consolidated OTel ES IndexNameProviders

### PR DESCRIPTION
**Pull request by joe-elliott**
## Which problem is this PR solving?
There were two `indexNameProvider`s with very similar code.  One for reading and one for writing.  These have been consolidated into one provider in the `internal/esutil` package

## Short description of the changes
- Consolidated `IndexNameProvider`s and tests

### CODE REVIEW

**Comment by joe-elliott** [comment]
@pavolloffay If you have a chance to look at this I would appreciate it.

File: cmd/opentelemetry/app/internal/esclient/index_name.go [hunk 1]
```diff
@@ -12,39 +12,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package esspanreader
+package esclient
```
Comment (pavolloffay) [inline cmd/opentelemetry/app/internal/esclient/index_name.go:15]: It should go to a different package than `esclient`. The `esclient` is dedicated only for the ES client used by Jaeger writer/reader.

**Review by pavolloffay — COMMENTED** [review]
It looks good but the package should be changed. I am fine to keep it in reader or exporter packages or move it to internal under something else than `esclient`.

File: cmd/opentelemetry/app/internal/esclient/index_name.go (hunk 1 above)
Comment (joe-elliott) [reply cmd/opentelemetry/app/internal/esclient/index_name.go]: moved to  `./internal/esutil`

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**